### PR TITLE
Add a recursive solution for @kaesonho

### PR DESCRIPTION
**Pull request by roderickhsiao**
@kaesonho, per request ...

### CODE REVIEW

File: javascript/algorithms/binarySearchRecursive.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        !element ||
+        target.constructor !== Array ||
+        target.legnth < 0) {
+        debug('ERROR, INVALID INPUT');
+        return -1;
+    }
+    var startIndex = start || 0;
+    var endIndex = end || target.length - 1;
+
+    while (startIndex <= endIndex) {
```
Comment (kaesonho) [inline javascript/algorithms/binarySearchRecursive.js]: why do we need the while loop?

Comment (roderickhsiao) [inline javascript/algorithms/binarySearchRecursive.js]: because I forget to change ...

**Comment by kaesonho** [comment]
looks good to me, do you like `recursive` ? ;)

**Comment by roderickhsiao** [comment]
:no_good:

**Pull request merged**